### ISSUE TITLE
A bug for assuming RESV can only be history matching wells

### ISSUE THREAD

**Issue by GitPaean** [OPM/opm-simulators#1866]
If WCONPROD is specified with RESV control,  the following assertion failure will be triggered. 

```
OPM-devel/debug/opm-simulators/opm/autodiff/SimFIBODetails.hpp:91: void Opm::SimFIBODetails::historyRates(const Opm::PhaseUsage&, const Opm::ProductionControls&, std::vector<double>&): Assertion `! p.prediction_mode' failed.
```

It should not be difficult to fix anyway. 

Possibly other related bugs can be found related to injectors.

**Comment by bska** [comment]
> the following assertion failure will be triggered.

The assertion seems fine, at least on the face of it. Ending up in something named `historyRates()` in the case of a well on prediction controls is surprising.

**Comment by GitPaean** [comment]
Yes. It looks even easier to fix.  A prediction producer should not enter this function.  Working for patch now.
